addons/rating/models/rating_mixin.py

```diff
@@ -79,6 +118,27 @@ def _compute_rating_stats(self):
             record.rating_count = mapping.get(record.id, {}).get('rating_count', 0)
             record.rating_avg = mapping.get(record.id, {}).get('rating_avg', 0)
 
+    def _search_rating_avg(self, operator, value):
+        if operator not in OPERATOR_MAPPING:
+            raise NotImplementedError('This operator %s is not supported in this search method.' % operator)
+        rating_read_group = self.env['rating.rating'].sudo().read_group(
+            [('res_model', '=', self._name), ('consumed', '=', True), ('rating', '>=', RATING_LIMIT_MIN)],
+            ['res_id', 'rating_avg:avg(rating)'], ['res_id'])
+        res_ids = [res['res_id'] for res in rating_read_group if OPERATOR_MAPPING[operator](float_compare(res['rating_avg'], value, 2), 0)]
```

Comment:
Do you prefer to write that on many lines? Like this:
```suggestion
        res_ids = [
            res['res_id']
            for res in rating_read_group
            if OPERATOR_MAPPING[operator](float_compare(res['rating_avg'], value, 2), 0)
        ]
```
🤔 😇

Reply:
Better as multiline in my opinion, but that is more personal taste ^^ .